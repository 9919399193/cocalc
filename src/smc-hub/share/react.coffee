###
Load react support for rendering.
###

async = require('async')

# Code for rendering react components to html.
ReactDOMServer = require('react-dom/server')

require('./jsdom-support')

# Uncomment for cc-in-cc dev benchmarking purposes.  This variable is already set
# by the Docker container when running in kubernetes.
## process.NODE_ENV="production"

require('smc-webapp/r_misc').SHARE_SERVER = true

# Load katex jQuery plugin.
require('smc-webapp/jquery-plugins/katex')

exports.react = (res, component, extra, viewer) ->
    res.type('html')
    t0 = new Date()
    stream = ReactDOMServer.renderToStaticNodeStream(component)
    stream.pipe(res)
    stream.once 'end', ->
        console.log("react: time to render and stream out: #{new Date() - t0}ms", extra)
