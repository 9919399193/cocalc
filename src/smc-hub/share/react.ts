/*
Load react support for rendering.
*/

// Code for rendering react components to html.
import * as ReactDOMServer from "react-dom/server";

import "./jsdom-support";

// Uncomment for cc-in-cc dev benchmarking purposes.  This variable is already set
// by the Docker container when running in kubernetes.
//# process.NODE_ENV="production"

import { set_share_server } from "smc-webapp/r_misc/share-server";

set_share_server(true);

// Load katex jQuery plugin.
require("smc-webapp/jquery-plugins/katex");

export function react(res, component, extra: any = ""): void {
  res.type("html");
  const t0 = new Date().valueOf();
  const stream = ReactDOMServer.renderToStaticNodeStream(component);
  stream.pipe(res);
  stream.once("end", () =>
    console.log(
      `react: time to render and stream out: ${new Date().valueOf() - t0}ms`,
      extra
    )
  );
}
