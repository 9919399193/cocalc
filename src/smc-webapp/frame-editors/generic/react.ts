// THIS SHOULD BE MOVED OUT OF frame-editors/


export const {
  React,
  ReactDOM,
  rclass,
  rtypes,
  Fragment,
  redux,
  Redux
} = require("smc-webapp/smc-react");

import * as ReactOrig from "react";

export const Component = ReactOrig.Component;

export type Rendered = ReactOrig.ReactElement<any> | undefined;
