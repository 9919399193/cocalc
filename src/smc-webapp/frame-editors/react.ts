export const {
  React,
  ReactDOM,
  rclass,
  rtypes,
  Fragment,
  redux,
  Redux
} = require("smc-webapp/smc-react");

import * as ReactOrig from "react";

export const Component = ReactOrig.Component;

export type Rendered = ReactOrig.ReactElement<any> | undefined;
