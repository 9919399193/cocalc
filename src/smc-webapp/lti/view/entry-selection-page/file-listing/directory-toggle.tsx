import * as React from "react";

interface Props {
  is_open: boolean;
  on_click: (was_open: boolean) => void;
}

export function DirectoryToggle({ is_open, on_click }: Props) {
  if (is_open) {
    return <span onClick={_ => on_click(is_open)}>▼ </span>;
  } else {
    return <span onClick={_ => on_click(is_open)}>► </span>;
  }
}
