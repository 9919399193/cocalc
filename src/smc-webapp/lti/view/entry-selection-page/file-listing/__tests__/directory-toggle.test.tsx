import * as React from "react";
import { shallow } from "enzyme";
import { DirectoryToggle } from "../directory-toggle";

describe("Render regressions", () => {
  test("is closed", () => {
    const rendered = shallow(<DirectoryToggle is_open={false} on_click={_ => {}} />);

    expect(rendered).toMatchSnapshot("Right pointing triangle");
  });

  test("is opened", () => {
    const rendered = shallow(<DirectoryToggle is_open={true} on_click={_ => {}} />);

    expect(rendered).toMatchSnapshot("Down pointing triangle");
  });
});

describe("Interactions", () => {
  test("Click", () => {
    const mock_on_click = jest.fn();
    const evt = new MouseEvent("click");
    const opened = true;
    const rendered = shallow(<DirectoryToggle is_open={opened} on_click={mock_on_click} />);

    rendered.simulate("click", evt);
    expect(mock_on_click.mock.calls.length).toBe(1);
    expect(mock_on_click.mock.calls[0][0]).toBe(opened);
  });
});
