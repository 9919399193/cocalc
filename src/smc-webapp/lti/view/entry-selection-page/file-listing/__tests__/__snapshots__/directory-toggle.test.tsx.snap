// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render regressions is closed: Right pointing triangle 1`] = `
<span
  onClick={[Function]}
>
  ► 
</span>
`;

exports[`Render regressions is opened: Down pointing triangle 1`] = `
<span
  onClick={[Function]}
>
  ▼ 
</span>
`;
